src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -1422,3 +1422,53 @@ function random_orientation_dag(
     end
     return g2
 end
+
+"""
+    bernoulli_graph(Λ)
+
+Given the symmetric matrix ``\\Lambda \\in [0,1]^{n \\times n}``, return a Bernoulli graph with ``n`` vertices. Each edge ``(i,j)`` exists with probability ``\\operatorname{Bernoulli}(\\Lambda[i,j])``.
+"""
+function bernoulli_graph(
+    Λ::Matrix{Float64};
+    rng::Union{Nothing,AbstractRNG}=nothing,
+    seed::Union{Nothing,Integer}=nothing,
+)
+    @assert size(Λ)[1] == size(Λ)[2] "The probability matrix must be a square matrix!"
```

Comment:
Using `@assert` in code is usually not a good idea, as the compiler might remove it. I would only use it in cases where it is absolutely clear, that the condition is always true, as a way of documentation.

From the Julia docs:
>  An assert might be disabled at various optimization levels. Assert should therefore only be used as a debugging tool and not used for authentication verification (e.g., verifying passwords), nor should side effects needed for the function to work correctly be used inside of asserts.

So we probably want to throw a proper exception here.

Reply:
Also, we ensure here, that the matrix is square, but not that it is symmetric. Maybe we should add a check for that, or we could also add to the documentation that we only check either the upper, or lower triangle of the matrix.